lib/forspell/reporter.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'pastel'
+require 'logger'
+require 'json'
+
+module Forspell
+  class Reporter
+    SUCCESS_CODE = 0
+    ERROR_CODE = 1
+    FORMAT = '%<file>s:%<line>i: %<text>s (suggestions: %<suggestions>s)'
+
+    def initialize(logfile:,
+                   verbose:,
+                   format:,
+                   group:)
+
+      FileUtils.touch(logfile) if logfile.is_a?(String)
+      @logger = Logger.new(logfile || STDOUT)
+      @logger.level = verbose ? Logger::INFO : Logger::WARN
+
+      @logger.formatter = proc do |_severity, _datetime, _progname, msg|
+        "#{msg}\n"
+      end
```

Comment:
Just stylistical -- you can `@logger.formatter = proc { |*, msg| "#{msg}\n" }`, looks "lighter".